tools/rpkg/R/unnest.R

```diff
@@ -0,0 +1,79 @@
+# Registered in .onLoad()
+unnest.tbl_duckdb_connection <- function(data,
+                                         cols,
+                                         ...,
+                                         keep_empty = FALSE,
+                                         ptype = NULL,
+                                         names_sep = NULL,
+                                         names_repair = "check_unique") {
+
+  if (keep_empty) {
+    stop("`keep_empty` unsupported.")
+  }
+  if (!is.null(ptype)) {
+    stop("`ptype` unsupported.")
+  }
+
+  # Side effect: check that tidyr is loaded, brings in rlang, vctrs, tidyselect
+  # and dplyr
+  pkg_method("unnest", "tidyr")
+  rlang::check_required(cols)
+
```

Comment:
Is this now causing that those "side effect" packages become hard dependencies (i.e. imports-dependencies)? Should we use `pkg_method()` for all required functions to keep them in suggested dependencies?

Reply:
No, all these packages stay suggested dependencies. The `pkg_method()` call fails if tidyr is not available (which is unlikely because this is a tidyr method...). Access via `::` is CRAN compatible, we only need to list the packages in "Suggests".